### PR TITLE
Update lambda

### PR DESCRIPTION
**Pull request by sfc-gh-pkommini**
Update lambda to latest code

### CODE REVIEW

**Review by sfc-gh-cnakhate — APPROVED** [review]

**Review by sfc-gh-rgaikwad — APPROVED** [review]

**Review by sfc-gh-afedorov — CHANGES_REQUESTED** [review]
don't undo the changes to Google API Client driver

File: lambda-code/drivers/process_https.py [hunk 1]
```diff
@@ -135,9 +139,18 @@ def process_row(
             )
             result = pick(req_results_path, response)
         except HTTPError as e:
+            response_body = e.read().decode()
+            content_type = e.headers.get('Content-Type')
             result = {
-                'error': f'{e.code} {e.reason}',
+                'error': 'HTTPError',
                 'url': next_url,
+                'status': e.code,
```
[the remark this answers is not in the document]
Comment (sfc-gh-pkommini) [reply lambda-code/drivers/process_https.py:147]: There is no status. I got a `mypy` type error.

File: lambda-code/drivers/process_google.py [hunk 2]
```diff
@@ -1,35 +1,28 @@
...
     resource_name,
     method,
     params,
-    service_account_info=None,
-    authorized_user_info=None,
     without_subject=False,
     subject=None,
     scopes=None,
 ):
-    if authorized_user_info is not None:
```
[the remark this answers is not in the document]
Comment (sfc-gh-pkommini) [reply lambda-code/drivers/process_google.py:20]: @sfc-gh-afedorov We'll have to review together and merge. I didn't mean to remove this. I just copied the entire code from `geff_aws/lambda-code` module we had.

[the remark this answers is not in the document]
Comment (sfc-gh-afedorov) [reply lambda-code/drivers/process_google.py:20]: oh, let's split this up into separate commits, then, so we make sure we understand what we are changing, then. I had been adding fixes to this and the `sfc-security` copy for awhile which will need to be merged here.

File: lambda-code/drivers/process_https.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfc-gh-afedorov) [reply lambda-code/drivers/process_https.py:147]: weird, iirc the documentation right, the error and response should have identical interfaces, would be curious to find out what the difference is, but not urgent.

**Comment by sfc-gh-afedorov** [comment]
esp since we do not have tests, let's re-make these changes in small chunks we can be confident aren't breaking anythign.

File: lambda-code/drivers/process_https.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfc-gh-pkommini) [reply lambda-code/drivers/process_https.py:147]: These are the [docs](https://docs.python.org/3/library/urllib.error.html)

[the remark this answers is not in the document]
Comment (sfc-gh-afedorov) [reply lambda-code/drivers/process_https.py:147]: ya, they're pretty vague with "Though being an exception [...] an HTTPError can can also function as [...] the same thing that urlopen() returns", but fwiw the implementation is true to that wording —

~~~
In [10]: try:
    ...:     response = urlopen(Request('https://api.abuseipdb.com/api/v2/check', method='POST'))
    ...: except Exception as e:
    ...:     response = e
    ...: 

In [11]: response
Out[11]: <HTTPError 405: 'Method Not Allowed'>

In [12]: response.code
Out[12]: 405

In [13]: response.status
Out[13]: 405
~~~

All good either way, interesting choice

**Pull request closed without merging**